### ISSUE TITLE
Unable to Refer to Clippy Lints on Beta

### ISSUE THREAD

**Issue by etaoins** [rust-lang/rust#54406]
My existing code has:

```rust
#[allow(unreadable_literal)]
```

Which produces this warning on the beta channel:
```
lint name `unreadable_literal` is deprecated and may not have an effect in the future. Also `cfg_attr(cargo-clippy)` won't be necessary anymore

note: #[warn(renamed_and_removed_lints)] on by default
help: change it to: `clippy::unreadable_literal`
```

However, changing it to `clippy::unreadable_literal` causes the following:
```
scoped lint `clippy::unreadable_literal` is experimental (see issue #44690)
```

On nightly `#![feature(tool_lints)]` can be used as an escape hatch but that's not an option on beta.

IMO neither Nightly nor Beta should generate warnings to change code to something that requires a feature flag. The existing code was valid for the stable features of the language.

**Comment by joelgallant** [comment]
The offending lint is here in beta: https://github.com/rust-lang/rust/blob/b7e44027a52cf7f2668c177582d0c80998b628b4/src/librustc/lint/levels.rs#L278

It appears to be different (and not "may not", but rather "will not") in nightly.

**Comment by etaoins** [comment]
cc #44690

**Comment by etaoins** [comment]
This should be fixed by #54870

**Comment by sdroege** [comment]
@etaoins This is still not optimal: it's not possible to silence the warning, and doing what the warning suggests makes compilation fail with current stable and older releases.

As such, running nightly clippy (and soon stable clippy) will always produce warnings unless the codebase drops support for an older toolchain version.

**Comment by Mark-Simulacrum** [comment]
I'm going to reopen this so that we can look and see if there's some warning-free way of doing this.
